queue_job/job.py

```diff
@@ -527,6 +527,14 @@ def perform(self):
             elif not self.max_retries:  # infinite retries
                 raise
             elif self.retry >= self.max_retries:
+                hook = f"{self.method_name}_on_max_retries_reached"
+                if hasattr(self.recordset, hook):
+                    recordset = self.recordset.with_context(
+                        job_uuid=self.uuid, exc_info=self.exc_info
+                    )
+                    getattr(recordset, hook)(*tuple(self.args))
+                    # Should be set_done after perform
+                    return
```

Comment:
I'm confused about that part, why return?
we still want to raise final FailedJobError
to set the job as failed
see [here](https://github.com/OCA/queue/blob/373cbca82b9dd942184e3926e1c67ca25b6aa373/queue_job/controllers/main.py#L135)

Reply:
Ah my bad, I misunderstood, I thought that the failed job would be retrieved and re-start